lib/utils/providers/locale_provider.dart

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022 The dahliaOS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import 'package:intl/locale.dart';
+import 'package:pangolin/generated/locale.dart';
+import 'package:pangolin/utils/other/preferences.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:yatl_flutter/yatl_flutter.dart';
+import 'package:yatl_gen/yatl_gen.dart';
+
+class _Providers {
```

Comment:
I know you took this from app_store but the name + the fact that it's not only locale stuff + the fact the name of the file suggests another thing upsets me, please fix (name the class _LocaleProviders or some other stuff, maybe we should reconsider the approach used for how Pangolin works in general)